### ISSUE TITLE
Error when running on large media directory

### ISSUE THREAD

**Issue by twslade** [BlueAcornInc/ba-media-sanitizer#1]
The issue here is that there are too many files open asynchronously. There are a couple of workarounds, but I wanted to get feedback from the team on how to deal with it. Error is show below. Also, here is a nice article on the issue http://stackoverflow.com/questions/8965606/node-and-error-emfile-too-many-open-files 

You will have to test this on a big media directory to encounter the issue...

![image](https://cloud.githubusercontent.com/assets/1391619/12873808/1623da4c-cd95-11e5-9cf3-734cba745458.png)

**Comment by SamTay** [comment]
Yeah, there are a bunch of issues with it right now, still trying to get first working state. I can remove the asynchronosity if this is a problem but will see what other people propose.

**Comment by twslade** [comment]
@SamTay any ideas here? :)

**Comment by SamTay** [comment]
I forgot this ever happened. Yeah, the idea is to use the first and accepted answer of the SO link you posted above :smile:
